### PR TITLE
This PR improves `GetPod` functions

### PR DESCRIPTION
**Pull request by a4abhishek**
Create a function GetPods which will be improves previous GetPod
function.
- It takes `pod-name prefix` and `namespace`
- It returns all the pods (slice of pod) which matches these criteria

Create a new GetPod function (because Previous GetPod function has become GetPods)
- It takes exact name of the pod (earlier it was taking prefix) and namespace
- It return the pod matching that criteria and/or err

Revise ReloadPod function according to current structure of the Pod

Signed-off-by: Abhishek Kashyap <abhishek.kashyap@mayadata.io>

### CODE REVIEW

File: utils/k8s/k8s.go [hunk 1]
```diff
@@ -141,7 +148,7 @@ func (k8s K8S) GetContainerStateInPod(pod core_v1.Pod, containerIndex int, timeo
 
 	for len(pod.Status.ContainerStatuses) <= containerIndex && time.Since(startTime) < timeout {
 		time.Sleep(time.Second)
-		pod, err = k8s.ReloadPod(pod)
+		pod, err = k8s.ReloadPod(*pod)
```
Comment (satyamz) [inline utils/k8s/k8s.go]: What if `pod` value is nil?

Comment (a4abhishek) [reply utils/k8s/k8s.go]: will handle it

**Comment by a4abhishek** [comment]
@satyamz check if it addresses that issue now.

File: utils/k8s/k8s.go [hunk 2]
```diff
@@ -125,12 +132,16 @@ func (k8s K8S) GetPodPhaseStr(pod core_v1.Pod) string {
 //    :param timeout: maximum time duration to get the container's state.
 //                       This method does not very strictly obey this param.
 //    :return: k8s.io/api/core/v1.ContainerState: state of the container.
-func (k8s K8S) GetContainerStateInPod(pod core_v1.Pod, containerIndex int, timeout time.Duration) (core_v1.ContainerState, error) {
+func (k8s K8S) GetContainerStateInPod(pod *core_v1.Pod, containerIndex int, timeout time.Duration) (core_v1.ContainerState, error) {
+	if pod == nil {
```
[the remark this answers is not in the document]
Comment (satyamz) [reply utils/k8s/k8s.go]: cool

**Pull request merged**